spec.bs

```diff
@@ -476,12 +515,43 @@ To <dfn>fetch bidding or scoring javascript</dfn> with given [=URL=] |url|:
 1. Let |headers| be |response|'s [=response/header list=].
 1. If [=header list/getting a structured field value=] "X-Allow-FLEDGE" from |headers| does not
   return true, return failure.
-1. Let |mimeType| be the result of [=header list/extracting a MIME type=] from |headers|.
-1. If |mimeType| is failure or not a [=JavaScript MIME type=], return failure.
-1. If |mimeType|'s [=MIME type/parameters=]["`charset`"] [=map/exists=], and is not "utf-8" or
-  "us-ascii", return failure.
+1. Let |headerMimeType| be the result of [=header list/extracting a MIME type=] from |headers|.
+1. Return failure if any of:
+  1. |headerMimeType| is failure.
+  1. |mimeType| is "`application/javascript`" and |headerMimeType| is not a [=JavaScript MIME type=].
+  1. |mimeType| is "`application/json`" and |headerMimeType| is not a [=JSON MIME type=].
+1. Let |mimeTypeCharset| be |headerMimeType|'s [=MIME type/parameters=]["`charset`"].
+1. Return failure if any of:
+  1. If |mimeTypeCharset| does not [=map/exist=], or |mimeTypeCharset| is "utf-8", and |body| is not
+    [=UTF-8=] encoded.
+  1. If |mimeTypeCharset| is "us-ascii", and |body| is not [=ascii string=].
 1. Return |body|.
 
+</div>
+
+<div algorithm="build trusted bidding signals url">
+
+To <dfn>build trusted bidding signals url</dfn> given a [=URL=] |signalsUrl|, an [=ordered set=] of
+[=strings=] |keys|, and an [=ordered set=] of [=strings=] |igNames|:
+1. Let |queryParamsList| be a new [=list/is empty|empty=] [=list=].
+1. [=list/Append=] "hostname=" to |queryParamsList|.
+1. [=list/Append=] (escape) [=this=]'s [=relevant settings object=]'s
+  [=environment/top-level origin=] to |queryParamsList|.
+1. If |keys| is not [=set/is empty|empty=]:
+  1. [=list/Append=] "&keys=" to |queryParamsList|.
+  1. [=list/Append=] the result of (escape) [=string/concatenate=] |keys| with separator set to ","
+    to |queryParamsList|.
+1. If |igNames| is not [=set/is empty|empty]:
+  1. [=list/Append=] "&interestGroupNames=" to |queryParamsList|.
+  1. [=list/Append=] the result of (escape) [=string/concatenate=] |igNames| with separator set to
+    "," to |queryParamsList|.
+1. TODO: Do we want to add [=auction config/seller experiment group id=]?

```

Comment:
@JensenPaul do we want to include this parameter in the spec?

Reply:
ya, I think so.